flytekit/models/task.py

```diff
@@ -370,19 +373,29 @@ def container(self):
         """
         return self._container
 
+    @property
+    def config(self):
+        """
+        Arbitrary dictionary containing metadata for parsing and handling custom plugins.
+        :rtype: dict[Text, T]
+        """
+        return self._config
+
     def to_flyte_idl(self):
         """
         :rtype: flyteidl.core.tasks_pb2.TaskTemplate
         """
-        return _core_task.TaskTemplate(
+        task_template = _core_task.TaskTemplate(
             id=self.id.to_flyte_idl(),
             type=self.type,
             metadata=self.metadata.to_flyte_idl(),
             interface=self.interface.to_flyte_idl(),
             custom=_json_format.Parse(_json.dumps(self.custom), _struct.Struct()) if self.custom else None,
             container=self.container.to_flyte_idl() if self.container else None,
             task_type_version=self.task_type_version,
+            config={k: v for k, v in self.config.items()} if self.config is not None else None,
```

Comment:
Add a comment here?  This looks like something someone might just revert to `self.config` one day.

Reply:
I added a test that will fail if it's not correctly set